src/NHibernate/Mapping/ByCode/IIdMapper.cs

```diff
@@ -1,20 +1,45 @@
 using System;
+using NHibernate.Mapping.ByCode.Impl;
 using NHibernate.Type;
+using NHibernate.Util;
 
 namespace NHibernate.Mapping.ByCode
 {
+	// 6.0 TODO: move into IIdMapper,
+	// and probably add an ITypeMapper for mutualizing the four Type methods with IElementMapper and IPropertyMapper
+	// (Note that there is no IKeyPropertyMapper to be concerned with, the KeyPropertyMapper use IPropertyMapper
+	// directly instead.)
+	public static class IdMapperExtensions
+	{
+		public static void Type<TPersistentType>(this IIdMapper idMapper) where TPersistentType : IIdentifierType
+		{
+			ReflectHelper
+				.CastOrThrow<IdMapper>(idMapper, "generic Type method")
+				.Type<TPersistentType>();
+		}
+
+		public static void Type<TPersistentType>(this IIdMapper idMapper, object parameters) where TPersistentType : IIdentifierType
+		{
+			ReflectHelper
+				.CastOrThrow<IdMapper>(idMapper, "generic parameterized Type method")
+				.Type<TPersistentType>(parameters);
+		}
+
+		public static void Type(this IIdMapper idMapper, System.Type persistentType, object parameters)
```

Comment:
Isn't it the only method that needs to be added to interface? I think it's better to   keep other methods as extensions and inline them.